[Term Entry] SwiftUI View Modifier: .background() #1510

matthewogtong opened the pull request:
<!--
👋 Hi, thanks for sending a PR to Codecademy Docs! 💖
Please fill out all fields below to ensure your PR is reviewed quickly.

**IMPORTANT**

If you would like to receive credit for your contribution to an entry, make sure to have your Codecademy user profile linked to your GitHub account:

1. Go to your Codecademy dashboard.
2. Click on your profile image in the top-right and then choose "Profile".
3. Then click "Edit Profile".
4. In the "GitHub Username" field, add your username (without the @).
5. Click "Save Changes"

Of course, you can opt not to do this and be listed as an "Anonymous contributor", instead. :)
-->

### Description

- Created a new entry for the SwiftUI ViewModifier `.background()` method
Closes #1510

### Type of Change

- Adding a new entry
- Updating the documentation

### Checklist

<!-- Please check ALL the boxes: -->

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

<!--
Having trouble with the PR checker? Here are some common issues and resolutions:

- verify_formatting is failing
  - run `yarn format path/to/markdown/file.md` or `yarn format:all` and commit the results
- verify_lint is failing
  - same as above
  - if verify_lint is still failing, running `yarn lint` locally should let you know what needs to be changed by hand
- test is failing
  - ensure any new markdown files have a `Title` and `Description` defined in their metadata
  - ensure any new markdown files only contain alphanumerics and dashes in their file names and have the same name as their parent directory
  - if that looks ok, running `yarn test` locally should let you know what the issue is
-->

SSwiniarski commented on content/swiftui/concepts/viewmodifier/terms/background/background.md:
```diff
@@ -0,0 +1,65 @@
+
+![SwiftUI ViewModifier Background](https://raw.githubusercontent.com/Codecademy/docs/main/media/swiftui-viewmodifier-background.png)
+
+> **Note**: Keep in mind that the `.background()` modifier can only be applied to a single view. If you want to apply a background to a group of views, you should consider using a container view, such as `ZStack`. Another tip is that you can use the `.opacity()` modifier along with `.background()` to set the transparency of the background.
```
Can we rewrite the note to remove the second person pronoun? (You)

matthewogtong replied:
Thanks for the review and feedback! I have updated the note to remove second person pronouns.

SSwiniarski reviewed (changes requested):
@matthewogtong I did a first review and left one comment.

SSwiniarski reviewed (approved):
@matthewogtong looks good for a second review.

yangc95 commented on content/swiftui/concepts/viewmodifier/terms/background/background.md:
````diff
@@ -0,0 +1,65 @@
+
+```swift
+var body: some View {
+        VStack {
+            Text("I have an Indigo background!")
+        }
+        .padding()
+        .foregroundColor(.white)
+        .background(Color.indigo)
+        .cornerRadius(12)
````
```suggestion
    VStack {
        Text("I have an Indigo background!")
    }
    .padding()
    .foregroundColor(.white)
    .background(Color.indigo)
    .cornerRadius(12)
```

yangc95 commented on content/swiftui/concepts/viewmodifier/terms/background/background.md:
````diff
@@ -0,0 +1,65 @@
+
+```pseudo
+View
+    .background(
+        _ content: View,
+        alignment: Alignment = .center,
+        blendMode: BlendMode? = nil,
+        ignoresSafeAreaEdges: Edge.Set = .init()
+    )
````
```suggestion
    .background(
        _ content: View,
        alignment: Alignment = .center,
        blendMode: BlendMode? = nil,
        ignoresSafeAreaEdges: Edge.Set = .init()
    )
```

matthewogtong replied:
Hi @yangc95, could you please clarify these suggested changes, I do not see any differences here.

yangc95 replied:
Please disregard this comment, the 4-spaces are correct

yangc95 commented on content/swiftui/concepts/viewmodifier/terms/background/background.md:
```diff
@@ -0,0 +1,65 @@
+And a few optional parameters:
+
+- `alignment`: The alignment of the background within the view.
+  > **Note**: The `alignment` parameter's default value is `center`, but other options include `.top`, `.bottom`, `.leading`, `.topTrailing`, `.bottomLeading`, and `.bottomTrailing`.
```
```suggestion
  > **Note**: The `alignment` parameter's default value is `.center`, but other options include `.top`, `.bottom`, `.leading`, `.topTrailing`, `.bottomLeading`, and `.bottomTrailing`.
```

yangc95 commented on content/swiftui/concepts/viewmodifier/terms/background/background.md:
```diff
@@ -0,0 +1,65 @@
+
+- `content`: The view to use as the background.
+
+And a few optional parameters:
```
```suggestion
Additionally, there are a few optional parameters:
```

yangc95 commented on content/swiftui/concepts/viewmodifier/terms/background/background.md:
```diff
@@ -0,0 +1,65 @@
+- `alignment`: The alignment of the background within the view.
+  > **Note**: The `alignment` parameter's default value is `center`, but other options include `.top`, `.bottom`, `.leading`, `.topTrailing`, `.bottomLeading`, and `.bottomTrailing`.
+- `blendMode`: The blend mode to use when rendering the background.
+- `ignoresSafeAreaEdges`: Whether the background should ignore safe area edges.
```
```suggestion
- `ignoresSafeAreaEdges`: A boolean indicating whether the background should ignore safe area edges.
```

yangc95 commented on content/swiftui/concepts/viewmodifier/terms/background/background.md:
````diff
@@ -0,0 +1,65 @@
+}
+```
+
+In the above example, we have a `Text` view displaying the string `"I have an Indigo background!"`. The `.padding()` modifier adds space around the text, while `.foregroundColor()` sets the text color to white.
````
```suggestion
In the above example, a `Text` view ia displaying the string `"I have an Indigo background!"`. The `.padding()` modifier adds space around the text, while `.foregroundColor()` sets the text color to white.
```

yangc95 commented on content/swiftui/concepts/viewmodifier/terms/background/background.md:
```diff
@@ -0,0 +1,65 @@
+
+![SwiftUI ViewModifier Background](https://raw.githubusercontent.com/Codecademy/docs/main/media/swiftui-viewmodifier-background.png)
+
+> **Note**: Keep in mind that the `.background()` modifier can only be applied to a single view. If a background is required for a group of views, it is recommended to use a container view, such as `ZStack`. Additionaly, the `.opacity()` modifier can be used along with `.background()` to adjust the transparency of the background.
```
```suggestion
> **Note**: Keep in mind that the `.background()` modifier can only be applied to a single view. If a background is required for a group of views, it is recommended to use a container view, such as [`ZStack`](https://www.codecademy.com/resources/docs/swiftui/views/zstack). Additionaly, the `.opacity()` modifier can be used along with `.background()` to adjust the transparency of the background.
```

yangc95 reviewed (commented):
@matthewogtong I did a second review. Please tag me once it's complete, thanks!

matthewogtong commented on content/swiftui/concepts/viewmodifier/terms/background/background.md:
```diff
@@ -0,0 +1,65 @@
+- `alignment`: The alignment of the background within the view.
+  > **Note**: The `alignment` parameter's default value is `center`, but other options include `.top`, `.bottom`, `.leading`, `.topTrailing`, `.bottomLeading`, and `.bottomTrailing`.
+- `blendMode`: The blend mode to use when rendering the background.
+- `ignoresSafeAreaEdges`: Whether the background should ignore safe area edges.
```
Done! I've added possible values for those parameters.

matthewogtong commented:
> @matthewogtong I did a second review. Please tag me once it's complete, thanks!

@yangc95 Thanks for the second review! I've addressed all the comments and feedback. Please let me know if there are any further changes I should make.

yangc95 commented on content/swiftui/concepts/viewmodifier/terms/background/background.md:
````diff
@@ -0,0 +1,67 @@
+```pseudo
+View
+    .background(
+        _ content: View,
````
```suggestion
        content,
```

yangc95 commented on content/swiftui/concepts/viewmodifier/terms/background/background.md:
```diff
@@ -0,0 +1,67 @@
+View
+    .background(
+        _ content: View,
+        alignment: Alignment = .center,
```
```suggestion
        alignment: .alignment,
```

yangc95 commented on content/swiftui/concepts/viewmodifier/terms/background/background.md:
```diff
@@ -0,0 +1,67 @@
+
+This method has one required parameter:
+
+- `content`: The view to use as the background.
```
```suggestion
- `Content`: The view to use as the background.
```

yangc95 commented on content/swiftui/concepts/viewmodifier/terms/background/background.md:
````diff
@@ -0,0 +1,67 @@
+}
+```
+
+In the above example, a `Text` view ia displaying the string `"I have an Indigo background!"`. The `.padding()` modifier adds space around the text, while `.foregroundColor()` sets the text color to white.
````
```suggestion
In the above example, a `Text` view is displaying the string `"I have an Indigo background!"`. The `.padding()` modifier adds space around the text, while `.foregroundColor()` sets the text color to white.
```

yangc95 commented on content/swiftui/concepts/viewmodifier/terms/background/background.md:
```diff
@@ -0,0 +1,67 @@
+
+The `.background()` modifier is then applied with the `Color.indigo` value passed in as the content parameter, to give the view an indigo background.
+
+Finally, a corner radius of 12 points is applied to the view using the `.cornerRadius()` modifier. This creates a view with an indigo background, white text, and rounded corners.
```
```suggestion
Finally, a corner radius of `12` points is applied to the view using the `.cornerRadius()` modifier. This creates a view with an indigo background, white text, and rounded corners.
```
